package.json

```diff
@@ -36,13 +36,16 @@
     "test-coverage:e2e": "c8 --config .c8rc.json --report-dir coverage/e2e/ --all npm run test:e2e",
     "test-coverage:e2e-sim": "c8 --config .c8rc.json --report-dir coverage/e2e-sim/ --all npm run test:e2e:sim",
     "test-coverage:spec": "c8 --config .c8rc.json --report-dir coverage/spec/ --all npm run test:spec",
-    "benchmark": "yarn benchmark:files 'packages/*/test/perf/**/*.test.ts'",
-    "benchmark:files": "NODE_OPTIONS='--max-old-space-size=4096 --loader=ts-node/esm' benchmark --config .benchrc.yaml --defaultBranch unstable",
+    "benchmark": "UV_THREADPOOL_SIZE=$(node -e 'console.log(require(`os`).availableParallelism())') yarn benchmark:files 'packages/*/test/perf/**/*.test.ts'",
+    "benchmark:files": "UV_THREADPOOL_SIZE=$(node -e 'console.log(require(`os`).availableParallelism())') NODE_OPTIONS='--max-old-space-size=4096 --loader=ts-node/esm' benchmark --config .benchrc.yaml --defaultBranch unstable",
     "release:create-rc": "node scripts/release/create_rc.mjs",
     "release:tag-rc": "node scripts/release/tag_rc.mjs",
     "release:tag-stable": "node scripts/release/tag_stable.mjs",
     "release:publish": "lerna publish from-package --yes --no-verify-access"
   },
+  "dependencies": {
+    "@chainsafe/blst": "^1.0.0"
```

Comment:
any reason this is added as a dependency here? I don't think prod dependencies make a lot of since in the top level package.json, as it's never installed outside of development

Reply:
Its a peerDependency of `@chainsafe/bls`. I looked at two options.
1) put it at the top as its required across many packages
2) put is in the individual `packages/*/package.json` next to bls

I'm open to either but its not a dev dep, its used during production.

Reply:
`@chainsafe/blst` is native only and relies on napi, some of our packages will be used in browser so it will break them most probably

Reply:
> as its required across many packages

The top level package.json is development only, all packages need to declare their runtime dependencies inside the respective package.json file, otherwise it won't be installed when you install the package, only if the whole monorepo is installed.

Option (2) seems good to me, we can declare it as dependency if we know the package does not require to support browser environments, e.g. cli package.

If a package should be browser compatible but we switch the implementation based on environment or whether or not blst is installed, the correct approach me seems to declare it as a peer dependency in the respective package.json file, and we might wanna declare it as optional ([bls package declares this already](https://github.com/ChainSafe/bls/blob/a5aa94c617b2abc26ff43c2f0ca7161eabe29cd6/package.json#L123-L130)).

> some of our packages will be used in browser so it will break them most probably

If we declare it at top level it won't break any packages as it will never be installed outside the monorepo.

Reply:
Thats a good point.  By leaving it at the root, the individual packages will not get the dep, only the lodestar repo/project will.  The consumer of the packages can add as a peerDep similarly.

Reply:
> will end up in the package node_modules folder and not the root node_modules folder where all packages will have access to it

If a dependency is required by multiple packages it will always be installed in root directory `node_modules`

Reply:
Seems like this will do the trick [ee21968](https://github.com/ChainSafe/lodestar/pull/6616/commits/ee219680df9c7d08763548b42150f3eec8b23b48)

Reply:
Updated.